src/journal/journal-file.c

```diff
@@ -351,7 +351,7 @@ JournalFile* journal_file_close(JournalFile *f) {
         if (f->post_change_timer) {
                 int enabled;
 
-                if (sd_event_source_get_enabled(f->post_change_timer, &enabled) >= 0)
+                if (f->fd >= 0 && sd_event_source_get_enabled(f->post_change_timer, &enabled) >= 0)
                         if (enabled == SD_EVENT_ONESHOT)
                                 journal_file_post_change(f);
```

Comment:
i think it would be a lot nicer and more readbale if the f->fd check would move into journal_file_post_change() (i.e. make it exit early if `f->fd < 0`). After all, this code here only deals with timer event sources, not with fds, only the function itself brings the fd into the context